examples_test/config_test.go

```diff
@@ -0,0 +1,144 @@
+		}
+	})
+}
+
+// TestHelloWorld_explicitYamlConfig is similar to TestHelloWorld_yamlConfig but uses HieraConfig
+// option to explicitly define what file to use.
+func TestHelloWorld_explicitYamlConfig(t *testing.T) {
+	configOptions := make(map[string]px.Value)
+	configOptions[hieraapi.HieraConfig] = types.WrapString(`testdata/hiera.yaml`)
+
+	// Initialize a Hiera session with the ConfigLookupKey as the top-level function and perform a lookup and
+	// the created configOptions.
+	hiera.DoWithParent(context.Background(), provider.ConfigLookupKey, configOptions, func(c px.Context) {
+		result := hiera.Lookup(hiera.NewInvocation(c, nil, nil), `hello`, nil, nil)
+		if result == nil || `yaml data says hello` != result.String() {
+			t.Fatalf("unexpected result %v", result)
+		}
+	})
+}
+
+// TestHelloWorld_explicitYamlConfigFile is similar to TestHelloWorld_yamlConfig but uses a combination of
+// HieraRoot and HieraConfigFileName option.
+func TestHelloWorld_explicitYamlConfigFile(t *testing.T) {
+	configOptions := make(map[string]px.Value)
+	configOptions[hieraapi.HieraRoot] = types.WrapString(`testdata`)
+	configOptions[hieraapi.HieraConfigFileName] = types.WrapString(`special.yaml`)
+
+	// Initialize a Hiera session with the ConfigLookupKey as the top-level function and perform a lookup and
+	// the created configOptions.
+	hiera.DoWithParent(context.Background(), provider.ConfigLookupKey, configOptions, func(c px.Context) {
+		result := hiera.Lookup(hiera.NewInvocation(c, nil, nil), `hello`, nil, nil)
+		if result == nil || `yaml special data says hello` != result.String() {
+			t.Fatalf("unexpected result %v", result)
+		}
+	})
+}
+
+// TestHelloWorld_yamlAndSemiHardWired uses the MuxLookupKey to inject two lookup_key functions. The ConfigLookupKey
+// that consults the yaml config and the sayHello.
+func TestHelloWorld_yamlAndSemiHardWired(t *testing.T) {
+	configOptions := make(map[string]px.Value)
+
+	// The LookupProvidersKey stores a go slice of Hiera "lookup_key" functions that serve as the top level functions.
+	configOptions[provider.LookupProvidersKey] = types.WrapRuntime([]hieraapi.LookupKey{provider.ConfigLookupKey, sayHello})
+
+	// HieraRoot tells the ConfigLookupKey where to search for the config file.
+	configOptions[hieraapi.HieraRoot] = types.WrapString(`testdata`)
+
+	// Initialize a Hiera session with the MuxLookupKey as the top-level function and perform a lookup and
+	// the created configOptions.
+	hiera.DoWithParent(context.Background(), provider.MuxLookupKey, configOptions, func(c px.Context) {
+		// A lookup of just "hello" should hit the first provider, the ConfigLookupKey.
+		result := hiera.Lookup(hiera.NewInvocation(c, nil, nil), `hello`, nil, nil)
+		if result == nil || `yaml data says hello` != result.String() {
+			t.Fatalf("unexpected result %v", result)
+		}
+
+		// A lookup of "howdy" is not found using the yaml configuration, so it hits the second provider, the sayHello.
```

Comment:
This is hard to understand just reading the code seems I need to go on a hunt for files...